connectivity/drivers/cellular/QUECTEL/BG96/QUECTEL_BG96_CellularStack.cpp

```diff
@@ -157,6 +158,40 @@ nsapi_error_t QUECTEL_BG96_CellularStack::socket_connect(nsapi_socket_t handle,
     return err;
 }
 
+nsapi_error_t QUECTEL_BG96_CellularStack::get_ip_address(SocketAddress *address)
+{
+    if (!address) {
+        return NSAPI_ERROR_PARAMETER;
+    }
+    _at.lock();
+
+    _at.cmd_start_stop("+QIACT", "?");
+    _at.resp_start("+QIACT:");
+
+    int32_t context_type = 0;
+
+    if (_at.info_resp()) {
+        _at.skip_param(); // ID
+        _at.skip_param(); // State
+
+        context_type = _at.read_int();
+        if (context_type == 1) {
+            _stack_type = IPV4_STACK;
+        } else if (context_type == 2) {
+            _stack_type = IPV6_STACK;
+        }
+
+        if (_at.read_string(_ip, PDP_IPV6_SIZE) != -1) {
```

Comment:
I guess this should only run if context_type != 0

Reply:
context_type is always 1 or 2, would only make sense to run if context_state = 1 (activated)

Reply:
BTW, read_string would return -1 if there is no such string, so I believe it is fine like this. The overridden get_ip_address is also implemented this way.